## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 437778`
Commit: `fc07cb429368eb276b15ff17a61889512214f2fd`

---
### `aarch64-darwin`
<details>
  <summary>:x: 10 packages failed to build:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>parsedmarc</li>
    <li>parsedmarc.dist</li>
    <li>python312Packages.awswrangler</li>
    <li>python312Packages.awswrangler.dist</li>
    <li>python313Packages.awswrangler</li>
    <li>python313Packages.awswrangler.dist</li>
    <li>python313Packages.opensearch-py</li>
    <li>python313Packages.opensearch-py.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python312Packages.opensearch-py</li>
    <li>python312Packages.opensearch-py.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.awswrangler</summary>
<pre>Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...

Traceback (most recent call last):
  File "/nix/store/x4mb2blmclhwpbh2qxsx207vnfqhpb3v-python3.12-pyproject-hooks-1.2.0/lib/python3.12/site-packages/pyproject_hooks/_impl.py", line 402, in _call_hook
    raise BackendUnavailable(
pyproject_hooks._impl.BackendUnavailable: Cannot import 'hatchling.build'

ERROR Backend 'hatchling.build' is not available.</pre>
</details>
<details>
<summary>python313Packages.opensearch-py</summary>
<pre>  File "/nix/store/9px7zzsfnlmj61yy2fpl6z14w93j05is-python3-3.13.6/lib/python3.13/asyncio/base_events.py", line 712, in run_until_complete
    self.run_forever()
  File "/nix/store/9px7zzsfnlmj61yy2fpl6z14w93j05is-python3-3.13.6/lib/python3.13/asyncio/base_events.py", line 683, in run_forever
    self._run_once()
  File "/nix/store/9px7zzsfnlmj61yy2fpl6z14w93j05is-python3-3.13.6/lib/python3.13/asyncio/base_events.py", line 2042, in _run_once
    handle._run()
  File "/nix/store/9px7zzsfnlmj61yy2fpl6z14w93j05is-python3-3.13.6/lib/python3.13/asyncio/events.py", line 89, in _run
    self._context.run(self._callback, *self._args)
  File "/private/tmp/nix-build-python3.13-opensearch-py-3.0.0.drv-0/source/test_opensearchpy/test_async/test_server_secured/test_security_plugin.py", line 45, in asyncSetUp
    self.client = await get_test_client(
  File "/private/tmp/nix-build-python3.13-opensearch-py-3.0.0.drv-0/source/opensearchpy/_async/helpers/test.py", line 36, in get_test_client
    await client.cluster.health(wait_for_status="yellow")
  File "/private/tmp/nix-build-python3.13-opensearch-py-3.0.0.drv-0/source/opensearchpy/_async/client/cluster.py", line 129, in health
    return await self.transport.perform_request(
  File "/private/tmp/nix-build-python3.13-opensearch-py-3.0.0.drv-0/source/opensearchpy/_async/transport.py", line 391, in perform_request
    status, headers_response, data = await connection.perform_request(
  File "/private/tmp/nix-build-python3.13-opensearch-py-3.0.0.drv-0/source/opensearchpy/_async/http_aiohttp.py", line 245, in perform_request
    await self._create_aiohttp_session()
  File "/private/tmp/nix-build-python3.13-opensearch-py-3.0.0.drv-0/source/opensearchpy/_async/http_aiohttp.py", line 367, in _create_aiohttp_session
    self.session = aiohttp.ClientSession(</pre>
</details>
</details>

--- a/pkgs/development/python-modules/opensearch-py/default.nix
+++ b/pkgs/development/python-modules/opensearch-py/default.nix
@@ -3,6 +3,7 @@
   stdenv,
   buildPythonPackage,
   fetchFromGitHub,
+  fetchpatch,
 
   # build-system
   setuptools,
@@ -39,6 +40,15 @@ buildPythonPackage rec {
     tag = "v${version}";
     hash = "sha256-IAEh+rB26Zqv7j5g2YIRZRCAtFbBngoh+w8Z4e2bY+M=";
   };
+
+  patches = [
+    # Remove delete event_loop fixture to fix test with pytest-asyncio 1.x
+    (fetchpatch {
+      name = "remove-delete-event-loop-fixture.patch";
+      url = "https://github.com/opensearch-project/opensearch-py/commit/2f9eeaad3f7bd38518b23a59659ccf02fff19577.patch";
+      hash = "sha256-ljg9GiXPOokrIRS+gF+W9DnZ71AzH8WmLeb3G7rLeK8=";
+    })
+  ];
 
   nativeBuildInputs = [ setuptools ];
 
@@ -83,7 +93,7 @@ buildPythonPackage rec {
     "test_basicauth_in_request_session"
     "test_callable_in_request_session"
   ]
-  ++ lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isx86) [
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
     # Flaky tests: OSError: [Errno 48] Address already in use
     "test_redirect_failure_when_allow_redirect_false"
     "test_redirect_success_when_allow_redirect_true"

